lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -114,12 +115,20 @@
    * Expert: Create a new instance that will cache at most <code>maxSize</code>
    * queries with at most <code>maxRamBytesUsed</code> bytes of memory, only on
    * leaves that satisfy {@code leavesToCache}.
+   *
+   * Also, clauses whose cost is {@code skipCacheFactor} times more than the cost of the top-level query
+   * will not be cached in order to not slow down queries too much.
    */
   public LRUQueryCache(int maxSize, long maxRamBytesUsed,
-      Predicate<LeafReaderContext> leavesToCache) {
+                       Predicate<LeafReaderContext> leavesToCache, float skipCacheFactor) {
     this.maxSize = maxSize;
     this.maxRamBytesUsed = maxRamBytesUsed;
     this.leavesToCache = leavesToCache;
+    if (skipCacheFactor >= 1 == false) { // NaN >= 1 evaluates true
```

Comment:
```suggestion
    if (skipCacheFactor >= 1 == false) { // NaN >= 1 evaluates false
```

I got it backward, sorry about that.